apps/mobile/src/components/widgets/collectibles/components/collectible-card-layout.tsx

```diff
@@ -0,0 +1,13 @@
+import { Box, BoxProps } from '@leather.io/ui/native';
+
+interface CollectiblesCardLayoutProps extends BoxProps {
```

Comment:
@kyranjamie : I'm no sure how to extend the shopify `Box`. 
```
export type { BoxProps } from '@shopify/restyle';
```

Doing this works (applying the props) but gives a type error `Generic type 
 requires between 1 and 2 type arguments.`


That's why I have been using `defaultProps` so I can more easily extend `Box`. 

See [issue here](https://github.com/leather-io/issues/issues/324). 

I will try and figure out how to do it without `defaultProps`. 

Maybe you have an idea on how to fix it?

Reply:
Maybe look in our UI lib `touchable-opacity.native.tsx`, I think @edgarkhanzadian did something with shopify props there that might be helpful?

Reply:
Thanks, I'll check it out. 50% of my work is `I love coding` and 50% is `I really hate typescript`

Reply:
[The deprecation docs](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops) have examples of this

<img width="1026" alt="image" src="https://github.com/user-attachments/assets/f2748cba-933a-4384-9953-cbc27fba66e1">

Default props are a runtime concern, so not sure what the types have to do with this?

```
function CollectiblesCardLayout({ 
  children, 
  somePropThatNeedsADefault = false, 
  ...props 
}: CollectiblesCardLayoutProps) {ts
```

Reply:
I was setting default props as a way to extend the shopify theme components. I will try what Fara suggested in `touchable-opacity.native.tsx` as it seems like that's what I need

Reply:
That worked, thanks @fbwoolf 👏